jerry-port/default/default-date.c

```diff
@@ -57,30 +69,59 @@ static double FileTimeToUnixTimeMs (FILETIME ft)
  *         available. Otherwise, returns 0, assuming UTC time.
  */
 double jerry_port_get_local_time_zone_adjustment (double unix_ms,  /**< ms since unix epoch */
-                                                  bool is_utc)  /**< is the time above in UTC? */
+                                                  bool is_utc)  /**< is the time above in UTC, */
 {
 #ifdef _WIN32
-  FILETIME fileTime, localFileTime;
-  SYSTEMTIME systemTime, localSystemTime;
-  ULARGE_INTEGER time, localTime;
+  FILETIME utcFileTime, localFileTime;
+  SYSTEMTIME utcSystemTime, localSystemTime;
+  bool timeConverted = false;
+
+  /**
+   * If the time is earlier than the date 1601-01-02, then always using date 1601-01-02 to
+   * query time zone adjustment, this date(1601-01-02) will make sure both utc/local
+   * time succeed with Win32 API, using date(1601-01-01) may leading win32 api
+   * failure, as after convert between local time/utc time, the time may earlier than
+   * date 1601-01-01 in utc time, that's exceed the FILETIME representation range.
```

Comment:
suggestion, edited for language: If the time is earlier than the date 1601-01-02, then always using date 1601-01-02 to query time zone adjustment. This date (1601-01-02) will make sure both UTC and local time succeed with Win32 API. The date 1601-01-01 may lead to a win32 api failure, as after converting between local time and utc time, the time may be earlier than 1601-01-01 in UTC time, that exceeds the FILETIME representation range.

Reply:
Thanks